### PR TITLE
Add TextView listeners extensions

### PR DESCRIPTION
**Pull request by hluhovskyi**
Add extensions for TextView listeners which can be used in the following way:

```kotlin
val editText: EditText = ...
editText.doAfterChanged { s: Editable -> }
editText.doBeforeChanged { s, start, count, after ->  }
editText.doOnChanged { s, start, before, count ->  }
```

### CODE REVIEW

File: src/main/java/androidx/widget/TextView.kt [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+        override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {
+            beforeChanged?.invoke(s, start, count, after)
+        }
+
+        override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
+            onChanged?.invoke(s, start, before, count)
+        }
+    }
+    addTextChangedListener(watcher)
```
Comment (pchmielowski) [inline src/main/java/androidx/widget/TextView.kt]: What do you think about following?

`return object : TextWatcher { /* ... */ }.also { addTextChangedListener(it) }`

Comment (dewarder) [reply src/main/java/androidx/widget/TextView.kt]: For me variant with local variable looks clearer and it also follows convention which is used in `Animator`

Comment (Edward608) [reply src/main/java/androidx/widget/TextView.kt]: Which one do you think is better? `((Editable) -> Unit)? = null` or `(Editable) -> Unit = {}`?

Comment (JakeWharton) [reply src/main/java/androidx/widget/TextView.kt]: That is unrelated to these comments, but null.

File: src/main/java/androidx/widget/TextView.kt [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    beforeChanged: (s: CharSequence, start: Int, count: Int, after: Int) -> Unit
+): TextWatcher = addTextChangedListener(beforeChanged = beforeChanged)
+
+/**
+ * Add an action which will be invoked when text of this TextView has been changed.
+ *
+ * @return the [TextWatcher] added to the TextView
+ * @see [TextWatcher.onTextChanged]
+ */
+fun TextView.doOnChanged(
```
Comment (romainguy) [inline src/main/java/androidx/widget/TextView.kt]: I feel these methods should be called `doOnTextChanged`, `doBeforeTextChanged` and `doAfterTextChanged` for clarity and consistency with `addTextChangedListener` and the methods on `TextWatcher`.

Comment (dewarder) [reply src/main/java/androidx/widget/TextView.kt]: Makes sense. I am going to update PR soon.

**Comment by hluhovskyi** [comment]
@JakeWharton @romainguy updated PR to use `androidx.core` package

**Comment by EzraLopez** [comment]
What work is left here? Anyway I could help?

**Comment by JakeWharton** [comment]
Text watcher-based listeners have been merged in http://r.android.com/720364

**Pull request closed without merging**